src/components/API.js

```diff
@@ -0,0 +1,77 @@
+export default class Api {
+  constructor(baseUrl, headers) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleServerResponse(res) {
```

Comment:
It's great to have a method for processing the server response, so there are no code duplication for that